components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusHelpers.kt

```diff
@@ -29,10 +50,28 @@ interface GleanPlumbInterface {
  *
  * It should also provide a similar function for String substitution, though this scheduled for EXP-2159.
  */
-class GleanPlumbMessageHelper(
+class NimbusMessagingHelper(
     private val targetingHelper: NimbusTargetingHelperInterface,
     private val stringHelper: NimbusStringHelperInterface,
-) : NimbusStringHelperInterface by stringHelper, NimbusTargetingHelperInterface by targetingHelper
+    private val cache: MutableMap<String, Boolean> = mutableMapOf(),
+) : NimbusStringHelperInterface by stringHelper, NimbusTargetingHelperInterface, NimbusMessagingHelperInterface {
+
+    override fun evalJexl(expression: String): Boolean =
+        cache.getOrPut(expression) {
+            targetingHelper.evalJexl(expression)
+        }
+
+    override fun clearCache() = cache.clear()
+
+    override fun destroy() {
+        if (targetingHelper is FFIObject) {
+            targetingHelper.destroy()
+        }
+        if (stringHelper is FFIObject) {
+            stringHelper.destroy()
+        }
+    }
```

Comment:
Are there tests covering these methods?

Reply:
The `evalJexl` method is already tested in the [`NimbusMessagingTest`](https://github.com/mozilla/application-services/pull/6080/files#diff-e06c0ae9c4a583c0128b90d539310b186bdb62fe6295ba027d35a3dbe725d897).

Knowing this, @jeddai is happy to disregard.